frontend/public/components/factory/list-page.tsx

```diff
@@ -127,12 +127,16 @@ export const ListPageWrapper: React.FC<ListPageWrapperProps> = (props) => {
   } = props;
   const dispatch = useDispatch();
   const memoizedIds = useDeepCompareMemoize(reduxIDs);
+  const [hasFilter, changeFilter] = React.useState(false);
 
   React.useEffect(() => {
-    if (nameFilter) {
+    if (!hasFilter && nameFilter) {
+      changeFilter(true);
+    }
+    if (hasFilter) {
       memoizedIds.forEach((id) => dispatch(filterList(id, 'name', { selected: [nameFilter] })));
```

Comment:
I think we could probably accomplish similar results without adding state:
```suggestion
      memoizedIds.forEach((id) => dispatch(filterList(id, 'name', { selected: ...(nameFilter ? [nameFilter] : []) })));
```

This is untested but should clear the name filter if it is empty.

Reply:
I kept the previous solution because calling filterList when nameFilter does not exist causes other pages to break.